common/testcontainers/testcontainers.go

```diff
@@ -0,0 +1,175 @@
+package testcontainers
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/scroll-tech/go-ethereum/ethclient"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// TestcontainerApps testcontainers struct
+type TestcontainerApps struct {
+	postgresContainer *postgres.PostgresContainer
+	l1GethContainer   *testcontainers.DockerContainer
+	l2GethContainer   *testcontainers.DockerContainer
+
+	// common time stamp.
+	Timestamp int
+}
+
+// NewTestcontainerApps returns new instance of TestcontainerApps struct
+func NewTestcontainerApps() *TestcontainerApps {
+	timestamp := time.Now().Nanosecond()
+	return &TestcontainerApps{
+		Timestamp: timestamp,
+	}
+}
+
+// StartPostgresContainer starts a postgres container
+func (t *TestcontainerApps) StartPostgresContainer() error {
+	if t.postgresContainer != nil && t.postgresContainer.IsRunning() {
+		return nil
+	}
+	postgresContainer, err := postgres.RunContainer(context.Background(),
+		testcontainers.WithImage("postgres"),
+		postgres.WithDatabase("test_db"),
+		postgres.WithPassword("123456"),
+		testcontainers.WithWaitStrategy(
+			wait.ForLog("database system is ready to accept connections").WithOccurrence(2).WithStartupTimeout(5*time.Second)),
+	)
+	if err != nil {
+		log.Printf("failed to start postgres container: %s", err)
+		return err
+	}
+	t.postgresContainer = postgresContainer
+	return nil
+}
+
+// StartL1GethContainer starts a L1Geth container
+func (t *TestcontainerApps) StartL1GethContainer() error {
+	if t.l1GethContainer != nil && t.l1GethContainer.IsRunning() {
+		return nil
+	}
+	req := testcontainers.ContainerRequest{
+		Image:        "scroll_l1geth",
+		ExposedPorts: []string{"8546/tcp", "8545/tcp"},
```

Comment:
```suggestion
		ExposedPorts: []string{"8545/tcp"},
```

Reply:
I think ws port is not necessary, can try removing it.

Reply:
the same as below